### PR TITLE
chore: Add docs for namespace scoped and cluster scoped privileges

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**
This PR updates the docs w.r.t 
> /kind documentation

**What does this PR do / why we need it**:

This PR adds docs for the below features
 - Default Permissions for a namespace scoped Argo CD Installation
 - How to configure an Argo CD instance to cluster scoped
 - Default Permissions for a cluster scoped Argo CD Instance
 - How to extend the permissions of an Argo CD Instance to deploy additional resources
 - How can Argo CD deploy resources to different namespaces

**Have you updated the necessary documentation?**
* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**How to test changes / Special notes to the reviewer**:
- Install Mkdocs
- Run `mkdocs serve`

Screenshot 2023-04-04 at 4 54 12 PM

Screenshot 2023-04-04 at 4 54 59 PM

### CODE REVIEW

**Comment by iam-veeramalla** [comment]
@jaideepr97 can you please review this ?

File: docs/usage/deploy-to-different-namespaces.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Deploy resources to a different namespace
+
+To grant Argo CD the permissions to manage resources in multiple namespaces, we need to configure the namespace with a label `argocd.argoproj.io/managed-by` and the value being the namespace of the Argo CD instance meant to manage the namespace.
```
Comment (jaideepr97) [inline docs/usage/deploy-to-different-namespaces.md:3]: can we rephrase this as "...value being the namespace of the managing Argo CD instance?" I think its a little less confusing that way

**Review by jaideepr97 — APPROVED** [review]
LGTM, thanks @iam-veeramalla
